satpy/modifiers/angles.py

```diff
@@ -0,0 +1,266 @@
+        # actually compute the storage to zarr
+        da.compute(new_res)
+
+
+def cache_to_zarr(
+        cache_config_key: str,
+        uncacheable_arg_types=(SwathDefinition, xr.DataArray, da.Array),
+        sanitize_args_func: Callable = None,
+) -> Callable:
+    """Decorate a function and cache the results as a zarr array on disk.
+
+    Note: Only the dask array is cached. Currently the metadata and coordinate
+    information is not stored.
+
+    """
+    # TODO: Consider cache management (LRU)
+    def _decorator(func: Callable) -> Callable:
+        zarr_cacher = ZarrCacheHelper(func,
+                                      cache_config_key,
+                                      uncacheable_arg_types,
+                                      sanitize_args_func)
+        wrapper = update_wrapper(zarr_cacher, func)
+        return wrapper
+    return _decorator
+
+
+def _hash_args(*args):
+    import json
+    hashable_args = []
+    for arg in args:
+        if isinstance(arg, (xr.DataArray, da.Array, SwathDefinition)):
+            continue
+        if isinstance(arg, AreaDefinition):
+            arg = hash(arg)
+        elif isinstance(arg, datetime):
+            arg = arg.isoformat(" ")
+        hashable_args.append(arg)
+    arg_hash = hashlib.sha1()
+    arg_hash.update(json.dumps(tuple(hashable_args)).encode('utf8'))
+    return arg_hash.hexdigest()
+
+
+def _sanitize_observer_look_args(*args):
+    new_args = []
+    for arg in args:
+        if isinstance(arg, datetime):
+            new_args.append(STATIC_EARTH_INERTIAL_DATETIME)
+        elif isinstance(arg, (float, np.float64, np.float32)):
+            # round floating point numbers to nearest tenth
+            new_args.append(round(arg, 1))
+        else:
+            new_args.append(arg)
+    return new_args
+
+
+def _geo_dask_to_data_array(arr: da.Array) -> xr.DataArray:
+    return xr.DataArray(arr, dims=('y', 'x'))
+
+
+def get_angles(data_arr: xr.DataArray) -> tuple[xr.DataArray, xr.DataArray, xr.DataArray, xr.DataArray]:
+    """Get sun and satellite angles to use in crefl calculations.
+
+    Note that this function can benefit from the ``satpy.config`` parameters
+    :ref:`cache_lonlats <config_cache_lonlats_setting>` and
+    :ref:`cache_sensor_angles <config_cache_sensor_angles_setting>`
+    being set to ``True``.
+
+    """
+    sun_angles = _get_sun_angles(data_arr)
+    sat_angles = _get_sensor_angles(data_arr)
+    # sata, satz, suna, sunz
+    return sat_angles + sun_angles
+
+
+def get_satellite_zenith_angle(data_arr: xr.DataArray) -> xr.DataArray:
+    """Generate satellite zenith angle for the provided data.
+
+    Note that this function can benefit from the ``satpy.config`` parameters
+    :ref:`cache_lonlats <config_cache_lonlats_setting>` and
+    :ref:`cache_sensor_angles <config_cache_sensor_angles_setting>`
+    being set to ``True``.
+
+    """
+    satz = _get_sensor_angles(data_arr)[1]
+    return satz
+
+
+@cache_to_zarr("cache_lonlats")
```

Comment:
Suggestion for the name of the decorator: `cache_to_zarr_if`. This way, the signification of the argument is made clear.

Reply:
To me the additional `_if` would indicate a conditional, and the supplied argument isn't a conditional but the key what to cache.

Reply:
@pnuu to clarify, the key provided is the config parameter name that is the boolean of whether or not to cache. The key used in caching is based on the name of the function being decorated. So with that said, what do you think about `_if`?